text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,434 @@
+  numberOfWorkers: 50
+});
+```
+
+### Required, Optional, and Overridable Parameters
+
+Each of these parameters are documented in [Glue's public documentation](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api.html);
+this table lists the parameters that will or will not be required, optional,
+and/or overridable for the Glue L2 construct.
+
+|Parameter|Required|Default Value|Overridable|
+|:---|:---|:---|:---|
+|ScriptLocation|Yes: S3 location|CommandName|yes S3 location|
+|NumberOfWorkers|No|10|Yes for ETL, STREAMING, RAY jobs|
+|Name|Yes|Default: Auto Generated name|Yes|
+|Role|Yes|None|Yes|
+|Description|No|None|Yes|
+|MaxConcurrentRuns|No|1|Yes|
+|DefaultArguments|No|None|Yes|
+|Connections|No|None|Yes|
+|MaxRetries|No|0|Yes|
+|Timeout|No|2 Days for non-streaming jobs|Yes|
+|SecurityConfiguration|No|None|Yes| |Tags|No|None|Yes|
+|GlueVersion|No|Default: 3.0 for ETL, RAY: 4.0|Yes|
+|Command name|No|ETL, PythonShell, Streaming and GlueRay|No|
+|Command Runtime|No|GlueRay: Ray2.4|No|
+|NonOverridableArguments|No|None|No|
+|ExecutionClass|No|STANDARD / FLEX|No|
+|Command PythonVersion|No|3 for ETL/Streaming, 3.9 for PythonShell|
+|LogUri|No|None|This option is not widely used|
+|MaxCapacity (Python Shell Jobs only)|No|0.0625|Yes|
+
+### Uploading scripts from the same repo to S3
+
+Similar to other L2 constructs, the Glue L2 will automate uploading / updating
+scripts to S3 via an optional fromAsset parameter pointing to a script
+in the local file structure. Developers will provide an existing S3 bucket and
+the path to which they'd like the script to be uploaded.
+
+```
+glue.ScalaSparkEtlJob(this, 'ScalaSparkEtlJob', {
+    script: glue.Code.fromAsset('bucket-name', 'local/path/to/scala-jar'),
+    className: 'com.example.HelloWorld',
+});
+```
+
+### Workflow Triggers
+
+In AWS Glue, developers can use workflows to create and visualize complex
+extract, transform, and load (ETL) activities involving multiple crawlers,
+jobs, and triggers. Standalone triggers are an anti-pattern, so we will
+only create triggers from within a workflow.
+
+Within the workflow object, there will be functions to create different
+types of triggers with actions and predicates. Those triggers can then be
+added to jobs.
+
+For all trigger types, the StartOnCreation property will be set to true by
+default, but developers will have the option to override it.
+
+1. **On Demand Triggers**
+
+On demand triggers can start glue jobs or crawlers. We’ll add convenience
+functions to create on-demand crawler or job triggers. The trigger method
+will take an optional description but abstract the requirement of an actions
+list using the job or crawler objects using conditional types.
+
+```
+myWorkflow = new glue.Workflow(this, "GlueWorkflow", {
+    name: "MyWorkflow";
+    description: "New Workflow";
+    properties: {'key', 'value'};
+});
+
+myWorkflow.createOnDemandTrigger(this, 'TriggerJobOnDemand', {
```

Comment:
Others can chime in if I'm being overly opinionated here, but I feel like CDK has a subtle standardization on avoiding action words like create/add/etc when we can. I think the api here should read `myWorkflow.onDemandTrigger`. 

And my opinion continues for the other APIs that utilize the word `create`

Reply:
Will update.